mods/carts/functions.lua

```diff
@@ -0,0 +1,226 @@
+function carts:get_sign(z)
+	if z == 0 then
+		return 0
+	else
+		return z / math.abs(z)
+	end
+end
+
+function carts:manage_attachment(player, status, obj)
+	if not player then
+		return
+	end
+	local player_name = player:get_player_name()
+	if default.player_attached[player_name] == status then
+		return
+	end
+	default.player_attached[player_name] = status
+
+	if status then
+		player:set_attach(obj, "", {x=0, y=6, z=0}, {x=0, y=0, z=0})
+		player:set_eye_offset({x=0, y=-4, z=0},{x=0, y=-4, z=0})
+	else
+		player:set_detach()
+		player:set_eye_offset({x=0, y=0, z=0},{x=0, y=0, z=0})
+	end
+end
+
+function carts:velocity_to_dir(v)
+	if math.abs(v.x) > math.abs(v.z) then
+		return {x=carts:get_sign(v.x), y=carts:get_sign(v.y), z=0}
+	else
+		return {x=0, y=carts:get_sign(v.y), z=carts:get_sign(v.z)}
+	end
+end
+
+function carts:is_rail(pos, railtype)
+	local node = minetest.get_node(pos).name
+	if node == "ignore" then
+		local vm = minetest.get_voxel_manip()
```

Comment:
If the rails set a meta value on placement then you can call `get_meta` on unloaded positions.to determine whether it is rail or not.